tests/rptest/redpanda_cloud_tests/omb_validation_test.py

```diff
@@ -378,10 +380,35 @@ def test_max_partitions(self):
         # just warn on the latency if above expected
         self._warn_metrics(benchmark.metrics, warn_validator)
 
-        # fail test if the latency is above expected including fudge factor
-        benchmark.check_succeed()
+        try:
+            # fail test if the latency is above expected including fudge factor
+            benchmark.check_succeed()
+        except Exception as e:
+            self.logger.info(
+                "Latency issues detected, going to check for latency spikes and retry"
+            )
 
+            latency_metrics = benchmark.check_succeed(
+                validate_metrics=False, return_latency_metrics=True)
+
+            # Check for spikes and determine if a retry is needed
+            retry_needed = self.redpanda.detect_spikes_by_percentile(
+                results=latency_metrics)
+
+            # Decide whether to mark the test as failed or to retry
+            if retry_needed:
+                self.logger.info(
+                    "Retrying test due to detected short-duration spikes.")
+                return True
+            else:
+                raise Exception(
+                    "Persistent latency issues detected without short-duration spikes."
+                )
```

Comment:
I suggest we add `from e` at the end

Reply:
Thank you, will add it

Reply:
New code pushed, resolving it since we have new logic now